packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -5571,8 +5572,8 @@ abstract class RenderObjectElement extends Element {
   /// Creates an element that uses the given widget as its configuration.
   RenderObjectElement(RenderObjectWidget widget) : super(widget);
 
-  @override
-  RenderObjectWidget get widget => super.widget as RenderObjectWidget;
+  /// A typed override of [Element.widget].
```

Comment:
Would be helpful to add a little color to this doc explaining when one should use `widget` and when `typedWidget` and why.

Reply:
Since I removed the public API, I wonder what we should do here. We could warn about casting, but ideally we'd just fix that and delete the typedWidget private API later

Reply:
What do you mean by "just fix that" - do you mean making it generic? Or do you see another way to avoid the private `typedWidget`s?

Reply:
I mean ideally we would make casts faster, and then we wouldn't need to do the work around. I'll update the documentation to clarify the casting performance